bson/src/test/unit/org/bson/codecs/jsr310/InstantCodecSpecification.groovy

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.jsr310
+
+import org.bson.BsonDocument
+import org.bson.BsonDocumentReader
+import org.bson.BsonDocumentWriter
+import org.bson.BsonReader
+import org.bson.codecs.DecoderContext
+import org.bson.codecs.EncoderContext
+import org.bson.codecs.configuration.CodecConfigurationException
+import spock.lang.IgnoreIf
+import spock.lang.Specification
+
+@IgnoreIf({ javaVersion < 1.8 })
+class InstantCodecSpecification extends Specification {
+
+    def 'should round trip Instant successfully'() {
+        given:
+        def codec = new InstantCodec()
+
+        when:
+        def writer = new BsonDocumentWriter(new BsonDocument())
+        writer.writeStartDocument()
+        writer.writeName('instant')
+        codec.encode(writer, instant, EncoderContext.builder().build())
+        writer.writeEndDocument()
+
+        then:
+        writer.getDocument().get('instant').isDateTime()
```

Comment:
For all the spec tests, assert on the actual value of the DateTime, not just that it's the correct Bson type.

Reply:
Done